syntax = "proto3";

// The greeting service definition.
service AirlineService {
  rpc GetAllFlights (AllFlightsRequest) returns (AllFlightsReply){}
  rpc GetFreeSeats (FreeSeatRequest) returns (FreeSeatReply){}
  rpc Reserve (ReserveRequest) returns (ReserveReply){}
  rpc ConfirmReserve (ConfirmReserveRequest) returns (ConfirmReserveReply){}
  rpc CancelReserve (CancelReserveRequest) returns (CancelReserveReply){}
  rpc GetAllSeats (AllSeatsRequest) returns (AllSeatsReply){}
}

message AllFlightsRequest{

}

message AllFlightsReply{
  string all_flights = 1;
}

message FreeSeatRequest {
  int32 flight_id = 1;
}

message FreeSeatReply {
  string message = 1;
}

message ReserveRequest{
  int32 flight_id = 1;
  int32 seat_number = 2;
}

message ReserveReply{

}

message ConfirmReserveRequest{
  int32 flight_id = 1;
  int32 seat_number = 2;
}

message ConfirmReserveReply{

}

message CancelReserveRequest{
  int32 flight_id = 1;
  int32 seat_number = 2;
}

message CancelReserveReply{

}

message AllSeatsRequest{
  int32 flight_id = 1;
}

message AllSeatsReply{
  string message = 1;
}